how-to/containerd-kata.md

```diff
@@ -0,0 +1,357 @@
+- `plugins.cri.containerd.runtimes.kata`: it is magic in containerd (ref [the doc here](https://github.com/containerd/containerd/tree/master/runtime/v2#binary-naming)), in which the dot connected string `io.containerd.kata.v2` will be translated to `containerd-shim-kata-v2`, i.e. the binary name of the kata implementation of shim v2 API;
+[Containerd Runtime V2 (Shim API)](https://github.com/containerd/containerd/tree/master/runtime/v2), the 
+- `plugins.cri.containerd.runtimes.katacli`: the `containerd-shim-runc-v1` will call `kata-runtime`, which is the legacy way deprecated by the `containerd-shim-kata-v2`:
+
+```toml
+    [plugins.cri.containerd]
+      no_pivot = false
+    [plugins.cri.containerd.runtimes]
+      [plugins.cri.containerd.runtimes.runc]
+         runtime_type = "io.containerd.runc.v1"
+         [plugins.cri.containerd.runtimes.runc.options]
+           NoPivotRoot = false
+           NoNewKeyring = false
+           ShimCgroup = ""
+           IoUid = 0
+           IoGid = 0
+           BinaryName = "runc"
+           Root = ""
+           CriuPath = ""
+           SystemdCgroup = false
+      [plugins.cri.containerd.runtimes.kata]
+         runtime_type = "io.containerd.kata.v2"
+      [plugins.cri.containerd.runtimes.katacli]
+         runtime_type = "io.containerd.runc.v1"
+         [plugins.cri.containerd.runtimes.katacli.options]
+           NoPivotRoot = false
+           NoNewKeyring = false
+           ShimCgroup = ""
+           IoUid = 0
+           IoGid = 0
+           BinaryName = "/usr/bin/kata-runtime"
+           Root = ""
+           CriuPath = ""
+           SystemdCgroup = false
+```
+
+Based on the discussion in [containerd/containerd#2916](https://github.com/containerd/containerd/pull/2916),
+the containerd configuration will have a field for runtime configuration files (likely to be`config_file`). However, 
+the field has not been available yet and the `shimv2` will search configuration file in the default paths 
+(same as `kata-runtime` cli): `/etc/kata-containers/configuration.toml`, `/usr/share/defaults/kata-containers/configuration.toml`. 
+And the document will be updated once the `config_file` field is available.
+
+If you want to specify a configure file now, here is a workaround. The `shimv2` accept an environment variable,
+`KATA_CONF_FILE` for config file path. Then we may create a shell script like this:
+
+```bash
+#!/bin/bash
+KATA_CONF_FILE=/etc/kata-containers/firecracker.toml containerd-shim-kata-v2
+```
+
+Name it as `/usr/local/bin/containerd-shim-katafc-v2` and reference it in the configuration of containerd:
+
+```
+      [plugins.cri.containerd.runtimes.kata-firecracker]
+         runtime_type = "io.containerd.katafc.v2"
+```
+
+#### Kata Containers as the runtime for untrusted workload
+
+For the without RuntimeClass support cases, we could use the legacy annotation way to support using Kata Containers
```

Comment:
Lines 210-212 suggested rewrite: 
For cases without RuntimeClass support, we can use the legacy annotation method to support using Kata Containers for an untrusted workload. With the following configuration, you can run trusted workloads with a runtime such as `runc` and then, run an untrusted workload with Kata Containers: